diff -urbB tfel-TFEL-4.2.0-25002f8861bcce1f83966e8c7557d3e6c0a141aa.orig/CMakeLists.txt tfel-TFEL-4.2.0-25002f8861bcce1f83966e8c7557d3e6c0a141aa/CMakeLists.txt
--- tfel-TFEL-4.2.0-25002f8861bcce1f83966e8c7557d3e6c0a141aa.orig/CMakeLists.txt	2023-11-30 16:21:31.000000000 +0100
+++ tfel-TFEL-4.2.0-25002f8861bcce1f83966e8c7557d3e6c0a141aa/CMakeLists.txt	2024-06-26 14:50:40.193965218 +0200
@@ -321,7 +321,7 @@
   message(STATUS "python include path   ${PYTHON_INCLUDE_DIRS}")
   message(STATUS "python libraries path ${PYTHON_LIBRARY_PATH}")
   message(STATUS "python library        ${PYTHON_LIBRARY}")
-  string(REGEX REPLACE "[a-z]+.*$" "" PYTHONLIBS_VERSION_CLEANED "${PYTHONLIBS_VERSION_STRING}")
+  string(REGEX REPLACE "[-a-z+]+.*$" "" PYTHONLIBS_VERSION_CLEANED "${PYTHONLIBS_VERSION_STRING}")
   find_package(PythonInterp ${PYTHONLIBS_VERSION_CLEANED} REQUIRED)
   message(STATUS "python interpreter    ${PYTHON_EXECUTABLE}")
   if(enable-numpy-support AND PYTHON_EXECUTABLE AND (NOT NUMPY_INCLUDE_DIRS))
